compiler/test/fail_compilation/call_function_type.d

```diff
@@ -0,0 +1,20 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/call_function_type.d(18): Error: expected 1 function arguments, not 0
+fail_compilation/call_function_type.d(19): Error: cannot call `int(int)(3)` at compile time
```

Comment:
This error is very strange. I'm surprised that you can call a function by using it's type. I would have expected that this errors out with "type int(int) is not callable; use a variable instead". Maybe I am misunderstanding the behavior here.

Reply:
Yes, the error is worded as a catch all, and not very specific.

Reply:
But it's not just the error. If the function is called at runtime then it works. That's even more surprising.

Reply:
Either way, that is no reason to not merge this. We can look to fix this in subsequent PRs.